astropy/table/info.py

```diff
@@ -121,3 +123,96 @@ def __call__(self, option='attributes', out=''):
+    clutter to that class.
+
+    Parameters
+    ----------
+    tbl : Table object
+        Input table
+    serialize_method : dict, str
+        Dict with key values of column names or types, or str
+
+    Returns
+    -------
+    None (context manager)
+    """
+    def get_override_sm(col):
+        """
+        Determine if the ``serialize_method`` str or dict specifies an
+        override of column presets for ``col``.  Returns the matching
+        serialize_method value or ``None``.
+        """
+        # If a string then all columns match
+        if isinstance(serialize_method, str):
+            return serialize_method
+
+        # If column name then return that serialize_method
+        if col.info.name in serialize_method:
+            return serialize_method[col.info.name]
+
+        # Otherwise look for subclass matches
+        for key in serialize_method:
+            if isclass(key) and isinstance(col, key):
+                return serialize_method[key]
+
+        return None
+
+    # Setup for the context block.  Set individual column.info.serialize_method
+    # values as appropriate and keep a backup copy.  If ``serialize_method``
+    # is None or empty then don't do anything.
+
+    if serialize_method:
+        # Original serialize_method dict, keyed by column name.  This only
+        # gets set if there is an override.
+        original_sms = {}
+
+        # Go through every column and if it has a serialize_method info
+        # attribute then potentially update it for the duration of the write.
+        for col in tbl.itercols():
+            if hasattr(col.info, 'serialize_method'):
+                override_sm = get_override_sm(col)
+                if override_sm:
+                    # Make a reference copy of the column serialize_method
+                    # dict which maps format (e.g. 'fits') to the
+                    # appropriate method (e.g. 'data_mask').
+                    original_sms[col.info.name] = col.info.serialize_method
+
+                    # Set serialize method for *every* available format.  This is
+                    # brute force, but at this point the format ('fits', 'ecsv', etc)
+                    # is not actually known (this gets determined by the write function
+                    # in registry.py).  Note this creates a new temporary dict object
+                    # so that the restored version is the same original object.
+                    col.info.serialize_method = {fmt: override_sm
+                                                 for fmt in col.info.serialize_method}
+
+    # Finally yield for the context block
+    try:
+        yield
+    finally:
+        # Teardown (restore) for the context block.  Be sure to do this even
+        # if an exception occurred.
+        if serialize_method:
+            for col in tbl.itercols():
+                # Go through every column and if it has a serialize_method info
```

Comment:
This comment is a copy/paste artefact, I think. At least I expected it to note that this would restore the original method.

Also, why not directly iterate over `name, original_value in original_sms.items()` and do `tbl[name].info.serialize_method = original_value`?

Reply:
Good idea.  I think I just did the copy/paste without thinking.